crrExistingObjects.js

```diff
@@ -223,13 +244,45 @@ function _listObjectVersions(bucket, VersionIdMarker, KeyMarker, cb) {
         Prefix: TARGET_PREFIX,
         VersionIdMarker,
         KeyMarker,
-    }, cb);
+    }, (err, data) => {
+        if (err) {
+            // retry only on Network errors and if the process is set to retry
+            // on errors.
+            if (err.code && err.code === 'NetworkingError' && RETRY_ON_ERROR) {
+                const errMsg = 'Error listing object versions for bucket: ' +
+                    `${bucket}, retrying request..`;
+                log.error(errMsg, { error: err });
+                return _listObjectVersions(bucket, VersionIdMarker,
```

Comment:
The aws sdk is using its own retry with backoff mechanism and will eventually return an error back to us, but we need one also, otherwise this could be running in a loop without really failing.

Reply:
Yes, the change works on top of backoff mechanism, the request was to keep retrying on network errors until a fair response is returned. Introduced an environment variable RETRY_ON_ERROR to handle this.

Reply:
A better name to describe it would be `RETRY_INDEFINITELY_ON_ERROR`, which is not an acceptable behavior.

Reply:
Indeed there's always a point in time where it's better to give up than keep retrying forever. Considering the use case, I think a reasonable default could be to give up after 15mn of retries since that could be caused by a rather serious network issue that requires attention from an admin.

For the backoff algorithm, you may use or get inspired from the backbeat one in lib/BackbeatTask.js

Reply:
Yes exactly. To illustrate, if some action needed to be taken to correct whatever issue is happening, failing is the best way to trigger an alert and let ops people know. Not failing is the best way to delay any useful response.

Reply:
Recursion is removed and retry (on errors) count is set to 100. `customBackOff` is modified to try these 100 requests in 5 minutes.